parsec/mca/device/device_gpu.c

```diff
@@ -597,9 +597,9 @@ parsec_device_memory_reserve( parsec_device_gpu_module_t* gpu_device,
     int rc;
     (void)eltsize;
 
-    size_t how_much_we_allocate;
+    size_t alloc_size;
     size_t total_mem, initial_free_mem;
-    uint32_t mem_elem_per_gpu = 0;
```

Comment:
Apart renaming and reordering the code, what is the real change here ? The shift for the local variable from uint32_t to size_t ? Why is this needed ?

Reply:
the test for how_much_we_allocate > initial_free_mem was done twice, in similar but not exactly the same way. The second time, around line 699, the use of int32_t would cause the result to be random (truncated) on some large memory systems.